de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/OpenApiUtil.xtend

```diff
@@ -0,0 +1,88 @@
+package de.fhdo.lemma.service.open_api
+
+import de.fhdo.lemma.data.datadsl.extractor.DataDslExtractor
+import de.fhdo.lemma.data.DataModel
+import org.eclipse.emf.ecore.EObject
+import java.nio.file.Files
+import java.nio.file.Paths
+import de.fhdo.lemma.service.ServiceModel
+import de.fhdo.lemma.servicedsl.extractor.ServiceDslExtractor
+import de.fhdo.lemma.technology.Technology
+import de.fhdo.lemma.technology.technologydsl.extractor.TechnologyDslExtractor
+import de.fhdo.lemma.data.DataFactory
+import de.fhdo.lemma.data.PrimitiveType
+
+/**
+ * This class collects _static_ utility methods for the OpenAPI plugin.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+final class OpenApiUtil {
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    static val DATA_FACTORY = DataFactory.eINSTANCE
+    static def String removeInvalidCharsFromName(String str) {
+        var ret = str;
+        ret = ret.replaceAll("[^a-zA-Z0-9_]", "");
```

Comment:
Strings are immutable in Java. It is therefore safe to code
```
val ret = str.replaceAll("[^a-zA-Z0-9_]", "");
```

In addition, Xtend does not require trailing semicolons. Please remove them here and everywhere else.